utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -0,0 +1,399 @@
+
+        // + 1 for call instruction.
+        let mut instructions = Vec::with_capacity(params.len() * 2 + results.len() + 1);
+        for processed_param in process_sys_call_params(params, self.config.params_config()) {
+            match processed_param {
+                ProcessedSysCallParams::Alloc => {
+                    let pages_to_alloc = self
+                        .unstructured
+                        .int_in_range(0..=mem_size_pages.saturating_sub(1))?;
+                    instructions.push(Instruction::I32Const(pages_to_alloc as i32));
+                }
+                ProcessedSysCallParams::Value {
+                    value_type,
+                    allowed_values,
+                } => {
+                    let is_i32 = match value_type {
+                        ValueType::I32 => true,
+                        ValueType::I64 => false,
+                        ValueType::F32 | ValueType::F64 => {
+                            panic!("gear wasm must not have any floating nums")
+                        }
+                    };
+                    let instr = if let Some(allowed_values) = allowed_values {
+                        if is_i32 {
+                            Instruction::I32Const(allowed_values.get_i32(self.unstructured)?)
+                        } else {
+                            Instruction::I64Const(allowed_values.get_i64(self.unstructured)?)
+                        }
+                    } else if is_i32 {
+                        Instruction::I32Const(self.unstructured.arbitrary()?)
+                    } else {
+                        Instruction::I64Const(self.unstructured.arbitrary()?)
+                    };
+
+                    instructions.push(instr);
+                }
+                ProcessedSysCallParams::MemoryArray => {
+                    let upper_limit = mem_size.saturating_sub(1);
+
+                    let pointer_beyond = self.unstructured.int_in_range(0..=upper_limit)?;
+                    let offset = self.unstructured.int_in_range(0..=pointer_beyond)?;
+
+                    instructions.push(Instruction::I32Const(offset as i32));
+                    instructions.push(Instruction::I32Const((pointer_beyond - offset) as i32));
+                }
+                ProcessedSysCallParams::MemoryPtrValue => {
+                    // Subtract a bit more so entities from `gsys` fit.
+                    let upper_limit = mem_size.saturating_sub(100);
+                    let offset = self.unstructured.int_in_range(0..=upper_limit)?;
+
+                    instructions.push(Instruction::I32Const(offset as i32));
+                }
+            }
+        }
+
+        instructions.push(Instruction::Call(call_indexes_handle as u32));
+        instructions.extend(results);
+
+        Ok(instructions)
+    }
+
+    fn insert_sys_call_instructions(&mut self, instructions: &[Instruction]) -> Result<()> {
```

Comment:
@DennisInSky 

This doesn't skip insertion into internal function which calls properly `gr_reservation_send`. if you don't mind, will implement that in a separate PR.

Reply:
done in #3036